src/goto-instrument/code_contracts.cpp

```diff
@@ -242,6 +259,70 @@ void code_contractst::code_contracts(
       apply_contract(goto_function.body, it);
 }
 
+void code_contractst::assert_ensures(
+  goto_programt &goto_program,
+  goto_programt::targett target)
```

Comment:
How about `code_function_callt` rather than `targett` and avoid the call to `get_function_call()` ?

Reply:
I addressed that by factoring out the assertion generation in a different function.